x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DataFrameAnalyticsTask.java

```diff
@@ -187,8 +187,8 @@ public void setFailed(Exception error) {
         });
     }
 
-    public void persistProgress() {
-        persistProgress(client, taskParams.getId(), () -> {});
+    public void persistProgress(Runnable runnable) {
+        persistProgress(client, taskParams.getId(), runnable);
     }
 
     // Visible for testing
```

Comment:
I think it would be beneficial to log the "persisted progress" object in the `debug` logger. This would probably require a `setonce` or something to grab the `progressToStore` object.

Reply:
We already log progress if we go on to persist it. I added the progress in the debug logging when we skip storage because it's the same as the previously stored progress.

Reply:
@dimitris-athanasiou I was actually thinking of logging on line 203: `LOGGER.debug("[{}] Successfully indexed progress document", jobId);`

Which progress document? There is no way to tie in the loglines the progress starting to be indexed and then the same one completing.

Reply:
Ah, yes. Good point. Will add that too.